analyzers/src/SonarAnalyzer.CSharp/Common/Walkers/CatchLoggingInvocationWalker.cs

```diff
@@ -54,9 +55,10 @@ public sealed class CatchLoggingInvocationWalker : SafeCSharpSyntaxWalker
         new LoggingInvocationDescriptor(Serilog, KnownType.Serilog_ILogger, true),
         new LoggingInvocationDescriptor(Serilog, KnownType.Serilog_Log, false));
 
-    public CatchLoggingInvocationWalker(SemanticModel model)
+    public CatchLoggingInvocationWalker(SemanticModel model, bool visitConditionals = true)
```

Comment:
I would make this explicit, removing the `= true`.
Also, when instantiating this I would do it by name for clarity, since it's a boolean:

```csharp
new CatchLoggingInvocationWalker(c.SemanticModel, visitConditionals: false/true);
```

Reply:
Not needed anymore. I extracted a new class for the new behavior.